js/react-tweek/src/withTweekKeysFactory.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { Component, ComponentType, Context } from 'react';
+import { GetPolicy, NotPreparedPolicy, TweekRepository } from 'tweek-local-cache';
+import isEqual from 'lodash.isequal';
+import omit from 'lodash.omit';
+import { ComponentEnhancer } from './typeUtils';
+
+export type WithTweekKeysOptions = {
+  onError?: (error: Error) => void;
+  getPolicy?: GetPolicy;
+}
+
+export interface WithTweekKeys {
+  (keyPropsMapping: {[s: string]: string}, options?: WithTweekKeysOptions): (BaseComponent: ComponentType<any>) => ComponentType<any>
+  <TTweekProps>(keyPropsMapping: Record<keyof TTweekProps, string>, options?: WithTweekKeysOptions): ComponentEnhancer<TTweekProps>
+}
+
+type WithTweekKeysState = { [s: string]: any };
+type WithTweekKeysRepoProps = { _tweekRepo: TweekRepository };
+
+export default (TweekContext: Context<TweekRepository>, prepare: (key: string) => void): WithTweekKeys =>
+  function (
+    keyPropsMapping: {[s: string]: string},
+    { getPolicy, onError }: WithTweekKeysOptions = {}
+  ) {
+    if (!getPolicy || getPolicy.notPrepared === NotPreparedPolicy.prepare) {
+      Object.values(keyPropsMapping).forEach(key => prepare(key));
+    }
+
+    return (BaseComponent: ComponentType<any>) => {
+      class WithTweekKeys extends Component<WithTweekKeysRepoProps, WithTweekKeysState> {
```

Comment:
Can we move WithTweekKeys to the top closure context and pass the other params (keyPropsMapping, baseComponent, getPolicy, onError) as  props in the render function (some of them can be grouped with something like subscribe: (repo)=> subscription.
The reason is to prevent react from creating another new type of component for every withTweekKeysFactory invocation (beside of the actual HOC)

Reply:
`keyPropsMapping` can't be moved if you want to prepare the keys before rendering, the rest can

Reply:
plus, isn't the whole point of an hoc is to create a new component? 
also, you're not supposed to use `withTweekKeysFactory`, it's exported when using `createTweekContext`